static/js/on-off-switch.js

```diff
@@ -16,31 +16,52 @@
  * OnOffSwitch Constructor (extends Thing).
  *
  * @param Object description Thing description object.
+ * @param {String} format 'svg' or 'html'.
  */
-var OnOffSwitch = function(description) {
+var OnOffSwitch = function(description, format) {
   this.base = Thing;
-  this.base(description);
+  this.base(description, format);
+  if (format == 'svg') {
+    // For now the SVG view is just a link.
+    return this;
+  }
   // Parse on property URL
   if (this.propertyDescriptions.on.href) {
     this.onPropertyUrl = new URL(this.propertyDescriptions.on.href, this.href);
   }
   this.updateStatus();
   this.element.addEventListener('click', this.handleClick.bind(this));
+  return this;
 };
 
 OnOffSwitch.prototype = Object.create(Thing.prototype);
 
 /**
- * HTML view for Thing.
+ * HTML view for on/off switch.
  */
-OnOffSwitch.prototype.view = function() {
+OnOffSwitch.prototype.htmlView = function() {
   return '<div class="thing on-off-switch">' +
          '  <div class="thing-icon"></div>' +
          '  <span class="thing-name">' + this.name + '</span>' +
          '</div>';
 };
 
 /**
+ * SVG view for on/off switch.
+ */
+Thing.prototype.svgView = function() {
```

Comment:
If this is the SVG view for the on/off switch shouldn't this be OnOffSwitch.prototype?

Reply:
Huh, yes. I thought I fixed this once.